Storage/Storage/Protocols/StorageType.swift

```diff
@@ -51,4 +53,16 @@ public protocol StorageType {
     /// Asynchronously performs a given block on the StorageType's queue.
     ///
     func perform(_ closure: @escaping () -> Void)
+
+    /// Create an `NSFetchedResultsController` using this `StorageType`.
+    ///
+    /// We generally do not allow direct access to `NSManagedObjectContext` as that is hidden
+    /// behind `StorageType`. However, `NSFetchedResultsController` requires an
+    /// `NSManagedObjectContext` to be instantiated. This method solves that problem. Consumers can
+    /// create an `NSFetchedResultsController` without illegally referencing an
+    /// `NSManagedObjectContext`.
+    ///
+    func createFetchedResultsController<ResultType>(fetchRequest: NSFetchRequest<ResultType>,
```

Comment:
we don't have the abstraction setup for Core Data implementation details like `NSFetchRequest` and `NSFetchedResultsController`, it would be nice if these details could be abstracted in the future so that this protocol doesn't depend on Core Data